kore/src/Kore/Repl/Data.hs

```diff
@@ -825,3 +856,118 @@ substituteAlias
     toString = \case
         SimpleArgument str -> str
         QuotedArgument str -> "\"" <> str <> "\""
+
+createOnePathClaim
+    :: Claim claim
+    => (claim, CommonStrategyPattern)
+    -> Rule.OnePathRule Variable
+createOnePathClaim (cl, cpatt) =
+    Rule.OnePathRule
+    $ Rule.RulePattern
+        { left = pattToLeftPatt cpatt
+        , right = claimToRightPatt cl
+        , requires = Predicate.makeTruePredicate
+        , ensures = claimToEnsures cl
+        , attributes = Default.def
+        }
+  where
+    pattToLeftPatt
+        :: CommonStrategyPattern
+        -> TermLike Variable
+    pattToLeftPatt =
+        toTermLike . fromJust . extractUnproven
+    claimToRightPatt
+        :: Claim claim
+        => claim
+        -> TermLike Variable
+    claimToRightPatt =
+        Rule.right . coerce
+    claimToEnsures
+        :: Claim claim
+        => claim
+        -> Predicate.Predicate Variable
+    claimToEnsures =
+        Rule.ensures . coerce
+
+conjOfOnePathClaims
+    :: [Rule.OnePathRule Variable]
+    -> Sort
+    -> TermLike Variable
+conjOfOnePathClaims opCls sort =
+    foldr
+        TermLike.mkAnd
+        (TermLike.mkTop sort)
+        $ fmap Rule.onePathRuleToPattern opCls
+
+generateInProgressOPClaims
+    :: Claim claim
+    => MonadState (ReplState claim) m
+    => m [Rule.OnePathRule Variable]
+generateInProgressOPClaims = do
+    gphs <- Lens.use lensGraphs
+    cls <- Lens.use lensClaims
+    let pairs =
+            ( \(x, y) -> ([x], y) )
```

Comment:
Not a big fan of these two lambdas here, and it's hard to follow what happens in this function. This particular one is `Bifunctor.first pure` which might be clearer? The second one is also `Bifunctor.first ((cls !!) . unClaimIndex)` although I'm not sure it's a lot better.

Maybe `terminalPatterns` needs to do this itself, and have the signature `Map ClaimIndex ExecutionGraph -> [(claim, TermLike)]` or something?